### ISSUE TITLE
Is it possible to integrate RealTimeSTT with LLM?

### ISSUE THREAD

**Issue by sangheonEN** [KoljaB/RealtimeSTT#133]
I want to integrate REALTIMESTT and LLM to more accurately identify the intent of the inferred TEXT and define accurate events with the identified intent.

For example, when controlling a robot in real time with voice recognition, the STT result must be very accurate to define an event with the inferred TEXT through STT. However, if LLM can be added to it, the speaker's intent can be identified robustly even with slightly less accurate TEXT inference. Are you working on a project that takes this into consideration?

In other words, I want to work on a project on REALTIME STT + LLM, and I would like to ask for some advice.

Can you recommend a reference or methodology that I can try?

**Comment by KoljaB** [comment]
I have many projects integrating RealtimeSTT with LLMs you could have a look at. For example, [here is a simple one](https://github.com/KoljaB/RealtimeSTT/blob/master/tests/openai_voice_interface.py) from the RealtimeSTT test folder. [This one here](https://github.com/KoljaB/LocalEmotionalAIVoiceChat) has a more sophisticated LLM integration. [And this here](https://github.com/KoljaB/Linguflex), my main project, is a personal assistant which also includes function calling.

**Comment by sangheonEN** [comment]
Thank you very much for your answer.

In addition, I have a question about combining STT and LLM!

I am using faster whisper Korean model, and I want to use huggingface's llama3-ko model (https://huggingface.co/gagagiga/Llama-3-Open-Ko-8B-Q4_K_M-GGUF/tree/main) for LLM Korean model.

Then, I will use the open source here https://github.com/KoljaB/LocalEmotionalAIVoiceChat, modify the config below to use ollama model,

elif config.llm_provider == "ollama":
from llm_ollama.llm_handler import LLMHandler

Is it okay to run the LLM model on the server with the ollama run model_name command and then run the main code?

Is there anything else I need to do?

**Comment by KoljaB** [comment]
> Is it okay to run the LLM model on the server with the ollama run model_name command and then run the main code?

That is correct. You first need to execute "ollama run model_name".

> Is there anything else I need to do?

Yes. You need to open llm_ollama/completion_params.json and put the model_name as "model" parameter there.

> And can I refer to that code for a simple service on a web server?

Sure you can. I would recommend [this more modern](https://github.com/KoljaB/RealtimeSTT/blob/master/server/stt_server.py) webserver code though. Btw creating a web server that performs STT and TTS and that also scales well horizontally is far from easy.

> And is it available only window os?

All basic techniques work in Ubuntu too, RealtimeSTT as well as RealtimeTTS. Maybe you want to switch from ollama to another LLM provider there, I think vllm would be the fastest.

>  but do I have to clone the valley git and keep adding and removing only the modified parts?
> Is there no way to easily update this work?

To merge your custom changes with updates to the RealtimeSTT package, fork the repository, make your changes locally, and track updates from the original source. When updates are available, merge them with your custom code. This way git helps automate the merging of updates with your custom changes, so you don’t need to manually copy and paste code. It handles most of the work, and you only resolve conflicts if necessary.
